ui/perfherder/graphs/TestDataModal.jsx

```diff
@@ -99,6 +99,49 @@ export default class TestDataModal extends React.Component {
     this.processOptions();
   }
 
+  addRelatedApplications = async params => {
+    const { relatedSeries: relatedSignature } = this.props.options;
+    const { errorMessages } = this.state;
+    let relatedTests = [];
+
+    const response = await PerfSeriesModel.getSeriesList(
+      relatedSignature.repository_name,
+      params,
+    );
+
+    if (!response.failureStatus) {
+      relatedTests = response.data.filter(signature => {
+        const differentApplications =
+          signature.application !== relatedSignature.application;
+        const similarTestNames =
+          this.removeSubstring(signature.application, signature.name) ===
+          this.removeSubstring(
+            relatedSignature.application,
+            relatedSignature.name,
+          );
+        const samePlatform = signature.platform === relatedSignature.platform;
+        const sameProject =
+          signature.projectName === relatedSignature.repository_name;
+
+        return (
+          differentApplications &&
+          similarTestNames &&
+          samePlatform &&
+          sameProject
+        );
+      });
+    } else {
+      errorMessages.push(response.data);
```

Comment:
Since you're using `response.data` twice you could use destructuring assignment: `const { data, failureStatus } = await PerfSeriesModel...`

Reply:
Changed.